metrics/metrics.go

```diff
@@ -0,0 +1,80 @@
+package metrics
+
+import (
+	"context"
+	"net"
+	"net/http"
+
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/prometheus"
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/metric/instrument/syncint64"
+	"go.opentelemetry.io/otel/sdk/metric"
+)
+
+var (
+	log = logging.Logger("metrics")
+)
+
+type Metrics struct {
+	Exporter *prometheus.Exporter
+
+	HttpLatency syncint64.Histogram
+
+	s *http.Server
+}
+
+func New(metricsAddr string) (*Metrics, error) {
+
+	var m Metrics
+
+	var err error
+	if m.Exporter, err = prometheus.New(prometheus.WithoutUnits()); err != nil {
+		return nil, err
+	}
+
+	provider := metric.NewMeterProvider(metric.WithReader(m.Exporter))
+	meter := provider.Meter("ipni/dhstore")
+
+	if m.HttpLatency, err = meter.SyncInt64().Histogram("ipni/dhstore/http_latency",
+		instrument.WithUnit("milliseconds"),
+		instrument.WithDescription("Latency of DHStore HTTP API")); err != nil {
+		return nil, err
+	}
+
+	m.s = &http.Server{
+		Addr:    metricsAddr,
+		Handler: metricsMux(),
+	}
+
+	return &m, nil
+}
+
+func (m *Metrics) RecordHttpLatency(ctx context.Context, t int64, method, path string, status int) {
```

Comment:
I recommend taking `time.Duration` instead of `t int64` here to encapsulate the time unit conversion.